server/commands/teamCreator.ts

```diff
@@ -59,11 +56,29 @@ async function teamCreator({
       team: authP.team,
       isNewTeam: false,
     };
-  }
-  // A team id was provided but no auth provider was found matching those credentials
-  // The user is attempting to log into a team with an incorrect SSO - fail the login
-  else if (id) {
-    throw InvalidAuthenticationError("incorrect authentication credentials");
+  } else if (teamId) {
+    // A team id was provided but no auth provider was found matching those credentials
+    // The user is attempting to log into a team with an external or personal SSO
+
+    // Find an AuthenticationProvider that matches the SSO type (google, slack, etc)
+    authP = await AuthenticationProvider.findOne({
+      where: { name: authenticationProvider.name, teamId },
+      include: [
+        {
+          model: Team,
+          as: "team",
+          required: true,
+        },
+      ],
+    });
```

Comment:
we still need to return the auth provider associated with the team that matched the SSO provider the user was using.

Reply:
`teamCreator` is doing a lot right now, and it doesn't feel like it's named properly at this point — but I think this is coherent to the original design of this command and general pattern.

a cleaner design might be to have separate commands handle the existing team and new team cases. we can do that refactor down the line if this gets any more complicated